conference/src/test/java/com/kyriba/conference/management/service/HallServiceImplTest.java

```diff
@@ -0,0 +1,184 @@
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.verify;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class HallServiceImplTest
+{
+  @Mock
+  private HallRepository hallRepository;
+
+  @InjectMocks
+  private HallServiceImpl hallService;
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+
+  @Test
+  public void findHall()
+  {
+    // given
+    final String hallName = "Hall 1";
+    final int places = 10;
+    final long id = 22L;
+    Hall hall = newHall(hallName, places);
+    doReturn(Optional.of(hall)).when(hallRepository).findById(eq(id));
+
+    // when
+    HallResponse result = hallService.findHall(id);
+
+    // then
+    assertThat(result).isNotNull();
+    assertThat(result.getName()).isEqualTo(hallName);
+    assertThat(result.getPlaces()).isEqualTo(places);
+  }
+
+
+  @Test
+  public void findHall_ifHallNotFoundThrowException()
+  {
+    final long id = 22L;
+    doReturn(empty()).when(hallRepository).findById(id);
+
+    thrown.expect(EntityNotFoundException.class);
+    thrown.expectMessage("Hall not found.");
+
+    hallService.findHall(id);
+  }
+
+
+  @Test
+  public void findAllHalls()
+  {
+    // given
+    Hall hall1 = newHall("Hall 1", 11);
+    Hall hall2 = newHall("Hall 2", 12);
+    doReturn(asList(hall1, hall2)).when(hallRepository).findAll();
+
+    // when
+    List<HallResponse> result = hallService.findAllHalls();
+
+    // then
+    assertThat(result).isNotNull()
+        .containsExactlyInAnyOrder(new HallResponse(hall1), new HallResponse(hall2));
+  }
+
+
+  @Test
+  public void findAllHalls_emptyListIfHallsNotFound()
+  {
+    doReturn(emptyList()).when(hallRepository).findAll();
+
+    List<HallResponse> result = hallService.findAllHalls();
+
+    assertThat(result).isNotNull().isEmpty();
+  }
+
+
+  @Test
+  public void createHall()
+  {
+    // given
+    final long id = 22L;
+    HallRequest hallRequest = new HallRequest();
+    hallRequest.setName("h1");
+    hallRequest.setPlaces(11);
+
+    Hall hallToCreate = new Hall(hallRequest);
+    Hall createdHall = new Hall(hallRequest);
+    createdHall.setId(id);
+
+    doReturn(createdHall).when(hallRepository).save(any(Hall.class));
+
+    // when
+    long createdHallId = hallService.createHall(hallRequest);
+
+    // then
+    assertThat(createdHallId).isEqualTo(id);
+    verify(hallRepository).save(refEq(hallToCreate));
+  }
+
+
+  @Test
+  public void updateHall()
+  {
+    // given
+    final long id = 22L;
+    Hall storedHall = newHall("Stored Hall", 11);
+    doReturn(Optional.of(storedHall)).when(hallRepository).findById(eq(id));
+
+    HallRequest hallRequest = new HallRequest();
+    hallRequest.setName("h1");
+    hallRequest.setPlaces(11);
+    Hall updatedHall = new Hall(hallRequest);
+
+    // when
+    hallService.updateHall(id, hallRequest);
+
+    // then
+    verify(hallRepository).save(refEq(updatedHall));
+  }
+
+
+  @Test
+  public void updateNonexistentHallThrowException()
```

Comment:
You are using given/when/then in some places, but not everywhere. This is strange.

Reply:
done